inc/checkout-steps.php

```diff
@@ -1007,6 +1007,10 @@ public function add_phone_localisation_address_formats( $formats ) {
 		// Bail if viewing order confirmation page
 		if ( function_exists( 'is_order_received_page' ) && is_order_received_page() ) { return $formats; }
 
+		if ( did_action( 'woocommerce_email_customer_details' ) ) {
+			return $formats;
+		}
```

Comment:
@alamgircsebd although this works for emails, it may not work to prevent duplicate phone numbers in other areas.

Perhaps we could check if the string `{phone}` is already added to the formats before adding it?

Reply:
Actually, for email, there are have different data for phone number. So we don't need again the phone number in formats data. Also, there will be bail only when WC mail send.

Reply:
Ah ok. Got it.